packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +655,126 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+                    totalLinesToFetch = endIndex - startIndex;
+                }
+            }
+        }
+
+        let fetchLinesRemainder = totalLinesToFetch;
+
+        while (fetchLinesRemainder > 0) {
+            let curLinesToFetch = bufferSize;
+
+            if (fetchLinesRemainder > bufferSize) {
+                fetchLinesRemainder-=bufferSize;
+            } else {
+                curLinesToFetch = fetchLinesRemainder;
+                fetchLinesRemainder = 0;
+            }
+
+            const tspClientResponse = await tspClient.fetchTableLines(traceUUID, outputId, QueryHelper.tableQuery(this.columnIds, fetchIndex, curLinesToFetch,
+                { [QueryHelper.REQUESTED_TABLE_COLUMN_IDS_KEY]: columnsIds }));
+            const lineResponse = tspClientResponse.getModel();
+            if (!tspClientResponse.isOk() || !lineResponse) {
+                return;
+            }
+            const model = lineResponse.model;
+            const lines = model.lines;
+            lines.forEach(line => {
+                const obj: any = {};
+                const cells = line.cells;
+                const columnIds = model.columnIds;
+
+                if (this.showIndexColumn) {
+                    obj[0] = line.index.toString();
+                }
+
+                cells.forEach((cell, index) => {
+                    const id = this.showIndexColumn ? columnIds[index] + 1 : columnIds[index];
+                    obj[id] = cell.content;
+                });
+
+                linesArray.push(obj);
+            });
+
+            for (let i=0;i<lines.length;i++) {
+                // Stores each csv row data
+                const csvrow = [];
+                for (let j=0;j<lines[i].cells.length;j++) {
+                    // Get the text data of each cell of
+                    // a row and push it to csvrow
+                    const cellContent = lines[i].cells[j].content;
+                    if (cellContent.includes(',')) {
+                        csvrow.push('"' + cellContent + '"');
+                    } else {
+                        csvrow.push(cellContent);
+                    }
+                }
+                // Combine each column value with comma
+                csv_data.push(csvrow.join(','));
+            }
+        }
+
+        // combine each row data with new line character
+        const tableString = csv_data.join('\n');
+
+        const link = document.createElement('a');
+        link.setAttribute('href', `data:text/csv;charset=utf-8,${encodeURIComponent(tableString)}`);
+        link.setAttribute('download', 'export');
+        link.style.display = 'none';
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+    }
+
+    protected showExportMenu(): React.ReactNode {
+        return <React.Fragment>
+            <ul>
+                <li className='drop-down-list-item' key={0} onClick={() => this.exportOutput('table')}>Export table to csv</li>
```

Comment:
The menu remains open after being clicked while the slow operation is ongoing. This can be really bad because there is no feedback to the user, if the user clicks again, there are 2, 3, .... loops in parallel exporting the full trace, it can bring the UI to a halt...

At the very least the menu should be closed. But even so, the user might open and click it again while the previous operation is still ongoing. I would have expected a progress dialog, maybe with a Cancel button.